network/address.go

```diff
@@ -67,7 +68,81 @@ func (a Address) NetworkAddress() string {
 		return ""
 	}
 	vals := strings.Split(string(a), typeAddressSep)
-	return vals[1]
+	validNetworkAddress := vals[1]
+
+	host, port, err := net.SplitHostPort(validNetworkAddress)
+	if err != nil {
+		return ""
+	}
+
+	// checking the validity of the IP address (both IPv4 or IPv6)
+	if net.ParseIP(host) != nil {
+		return validNetworkAddress
+	}
+
+	// if host is not a valid IP address (i.e. ParseIP returns nil),
+	// it must be a hostname (since at the beginning we check whether the address is valid
+	ipAddress, err := net.LookupHost(host)
+	if err != nil {
+		return ""
+	}
+
+	sort.Strings(ipAddress)
```

Comment:
Yes, we should limit the number of "hacks" like that that are only needed for testing. I would remove that line, and decide on a good way to test that function. Your proposal sounds fine to me, but it can be even simplified using the `Interface` approach mentioned below (the `TestResolver` could return an array of length one, so there would be no need to search through the array).